### PR TITLE
Add test to validate cert name selection logic

### PR DESCRIPTION
**Pull request by geofffranks**
- [x] Read the [Contributing document](../blob/-/.github/CONTRIBUTING.md).

Summary
---------------
Adds a test to ensure we don't break the [certificate selection logic that broke](https://github.com/cloudfoundry/routing-release/issues/401) when we removed the BuildNameToCertificate() call in [v0.293.0](https://github.com/cloudfoundry/routing-release/releases/tag/v0.293.0) and [v0.294.0](https://github.com/cloudfoundry/routing-release/releases/tag/v0.294.0). 

Backward Compatibility
---------------
Breaking Change? No

### CODE REVIEW

**Comment by geofffranks** [comment]
@maxmoehl when you get a chance can you review this + validate it catches the case we regressed on?

**Review by maxmoehl — APPROVED** [review]
Yes, this should ensure we don't see this issue again. Thank you very much!

File: router/router_test.go [hunk 1]
```diff
@@ -2240,6 +2240,34 @@ var _ = Describe("Router", func() {
 			})
 		})
 	})
+
+	FDescribe("cert selection", func() {
```
Comment (winkingturtle-vmw) [inline router/router_test.go:2244]: I think the "Focus" has be removed

Comment (maxmoehl) [reply router/router_test.go:2244]: Uhh sorry, didn't notice that. I'll fix it :)

Comment (maxmoehl) [reply router/router_test.go:2244]: see #409

**Pull request merged**